packages/geo/src/lib/directions/directions.component.ts

```diff
@@ -435,4 +443,17 @@ export class DirectionsComponent implements OnInit, OnDestroy {
       this.languageService
     );
   }
+
+  onToggleDirectionsControl(toggle: boolean) {
+    this.queryService.queryEnabled = !toggle;
+    if (toggle) {
+      this.interactions.map((interaction) =>
+        this.routesFeatureStore.layer.map.ol.addInteraction(interaction)
+      );
+    } else {
+      this.interactions.map((interaction) =>
+        this.routesFeatureStore.layer.map.ol.removeInteraction(interaction)
+      );
+    }
```

Comment:
Simplification possible:
```suggestion
      const ol = this.routesFeatureStore.layer.map.ol;
      this.interactions.map((interaction) =>
        toggle ? ol.addInteraction(interaction) : ol.removeInteraction(interaction);
      );
```

Reply:
Done. Good idea!